src/main/java/org/openlmis/requisition/web/BaseRequisitionController.java

```diff
@@ -244,21 +248,122 @@ protected void checkIfPeriodIsValid(Requisition requisition) {
     }
   }
 
-  protected ValidationResult validateRequisitionCanBeUpdated(Requisition requisitionToUpdate,
-                                                             Requisition requisition) {
+  ValidationResult validateRequisitionCanBeUpdated(Requisition requisitionToUpdate,
+      Requisition requisition) {
     return requisitionToUpdate.validateCanBeUpdated(new RequisitionValidationService(
         requisition, requisitionToUpdate,
         dateHelper.getCurrentDateWithSystemZone(),
         datePhysicalStockCountCompletedEnabledPredicate.exec(requisitionToUpdate.getProgramId())));
   }
 
-  protected void validateForStatusChange(Requisition requisition) {
-    getValidationResultForStatusChange(requisition)
-        .throwExceptionIfHasErrors();
+  ValidationResult validateRequisitionCanBeUpdated(Requisition requisitionToUpdate,
+      Requisition requisition, ProgramDto program) {
+    return requisitionToUpdate.validateCanBeUpdated(new RequisitionValidationService(
+        requisition, requisitionToUpdate,
+        dateHelper.getCurrentDateWithSystemZone(),
+        datePhysicalStockCountCompletedEnabledPredicate.exec(program)));
   }
 
-  protected ValidationResult getValidationResultForStatusChange(Requisition requisition) {
+  ValidationResult getValidationResultForStatusChange(Requisition requisition) {
     return requisition.validateCanChangeStatus(dateHelper.getCurrentDateWithSystemZone(),
         datePhysicalStockCountCompletedEnabledPredicate.exec(requisition.getProgramId()));
   }
+
+  Profiler getProfiler(String name, Object... entryArgs) {
+    extLogger.entry(entryArgs);
+
+    Profiler profiler = new Profiler(name);
+    profiler.setLogger(extLogger);
+
+    return profiler;
+  }
+
+  void stopProfiler(Profiler profiler, Object... exitArgs) {
+    profiler.stop().log();
+    extLogger.exit(exitArgs);
+  }
+
+  Requisition findRequisition(UUID requisitionId, Profiler profiler) {
```

Comment:
Can you remove code duplication from 3 methods below?

Reply:
I added one method that will handle this situation.